var/spack/repos/builtin/packages/mfem/package.py

```diff
@@ -1185,9 +1201,10 @@ def check_or_test(self):
         # Running 'make check' or 'make test' may fail if MFEM_MPIEXEC or
         # MFEM_MPIEXEC_NP are not set appropriately.
         if not self.run_tests:
-            # check we can build ex1 (~mpi) or ex1p (+mpi).
-            make("-C", "examples", "ex1p" if ("+mpi" in self.spec) else "ex1", parallel=False)
-            # make('check', parallel=False)
+            if "+examples" in self.spec:
+                # check we can build ex1 (~mpi) or ex1p (+mpi).
+                make("-C", "examples", "ex1p" if ("+mpi" in self.spec) else "ex1", parallel=False)
+                # make('check', parallel=False)
```

Comment:
The point of this code is to always make sure we can build and run one executable. We want to do this regardless of the setting of the variant `examples` -- the code should work in both cases. Please revert this change.